embassy-executor/src/raw/mod.rs

```diff
@@ -328,15 +328,20 @@ impl SyncExecutor {
         #[cfg(feature = "integrated-timers")]
         let alarm = unsafe { unwrap!(embassy_time_driver::allocate_alarm()) };
 
-        Self {
+        let this = Self {
             run_queue: RunQueue::new(),
             pender,
 
             #[cfg(feature = "integrated-timers")]
             timer_queue: timer_queue::TimerQueue::new(),
             #[cfg(feature = "integrated-timers")]
             alarm,
-        }
+        };
+
+        #[cfg(feature = "integrated-timers")]
+        embassy_time_driver::set_alarm_callback(this.alarm, Self::alarm_callback, &this as *const _ as *mut ());
```

Comment:
You're right... How acceptable is it for me to introduce a mandatory function? It is a breaking change for sure, but taking a critical section for no good reason per poll is what motivates me here.

Reply:
oh it takes *a critical section*? then I think we should do it, yes... :sweat_smile: 

this might be one of the reeasons why `integrated-timers` is so much slower comparatively. huh.